src/software/thunderscope/replay/test/replay_corruption_test.py

```diff
@@ -0,0 +1,190 @@
+                # this would happen 90 % of the time
+                log_entry = create_valid_log_entry(proto, proto_time)
+            else: 
+                # this would happen 10% of the time
+                log_entry = gen_log_entry_func(proto, proto_time)
+
+            ProtoLogger.write_to_logfile(log_file, bytes(log_entry, encoding="utf-8"))
+
+def make_replay_chunk(size_of_replay_chunk=1000):
+    """
+    Creating a replay chunks that is like the following: 
+    1. from 0 to 0.1 seconds, it all of the entries are valid
+    2. from 0.1 to 0.2 seconds, 10% of the proto cannot be decoded 
+    3. from 0.2 to 0.3 seconds, 10% of the proto are missing delimeter
+
+    :size_of_replay_chunks: the size of the replay chunks we want to make
+    """
+    # making sure previous test results doesn't affect the new test
+    shutil.rmtree(TMP_REPLAY_SAVE_PATH, ignore_errors=True)
+
+    replay_proto =  []
+    # we divide by three because we make three copies in the code below, so we have
+    # close to size_of_replay_chunk in the end, assuming integer rounding that is. 
+    for _ in range(int(size_of_replay_chunk/3)):
+        some_proto = create_random_proto()
+        replay_proto.append(some_proto)
+
+    # making the replay chunks 
+    # the first part of the replay chunks is from  0 to 0.1 seconds, all of the entries are valid
+    # the seconds part of the replay chunk is from 0.1 to 0.2 seconds, 10% of the proto cannot be decoded 
+    # the third part of the replay chunk is from 0.2 to 0.3 seconds, 10% of the proto are missing delimeter
+    make_part_replay_chunks(replay_proto, TMP_REPLAY_SAVE_PATH, 0.1, 0.0, create_valid_log_entry)
+    make_part_replay_chunks(replay_proto, TMP_REPLAY_SAVE_PATH, 0.1, 0.1, create_corrupt_log_entry)
+    make_part_replay_chunks(replay_proto, TMP_REPLAY_SAVE_PATH, 0.1, 0.2, create_missing_delimeter_log_entry)
+
+def cleanup():
+    """
+    cleaning up this test by deleting the replay files 
+    """
+    shutil.rmtree(TMP_REPLAY_SAVE_PATH)
+
+def create_test_player() -> ProtoPlayer:
+    """
+    Creating a protoplayer and setting a seed so that the generated 
+    replay file stays the same over time!
+    """
+
+
+    make_replay_chunk()
+
+    io = ProtoUnixIO()
+    player = ProtoPlayer(TMP_REPLAY_SAVE_PATH, io)
+    return player 
+
+def test_for_file_corruption():
+    """
+    testing whether or not the proto player could play corrupted zipfile! 
+    """
+    player = create_test_player()
+
+    # block until the player has finished playing!
+    # in total, this should play for around 0.4 seconds 
+    while player.is_it_playing(): 
+        time.sleep(0.1)
+
+    # asserting that no exception has occurred when the protobufs are being played
+    assert player.get_error_bit_flag() == NO_ERROR_FLAG
+
+    cleanup()
+
+if __name__ == "__main__":
+    """
+    Please read the header of this file for more context on what it is testing!
+    """
+    sys.exit(pytest.main([__file__, "-vv"]))
```

Comment:
nit: we should use `pytest_main` for consistency. We currently don't use it in all tests, but you can find examples of some of the other pytests using it.

Reply:
done